[WIP] Updating principles from Collab Cafe (#241)
sa-tre/satre-specification: docs/source/principles.md

  Any generated data should follow standards for provenance and transparency for audit trails.
- Different levels of observability may be needed for different users.
  Any data collected from an observability perspective should consider the needs of those who will use it, and minimise collection accordingly.
- There may be ethical and confidential issues to consider when implementing the observability principle.

(principle_standardisation)=

## Standardisation

### Statement

TREs should adhere to standards or well-known patterns wherever possible.

### Rationale

Standardisation makes it easier to design, operate, use and understand TREs, and reduces duplication of work.
This includes making TREs easier to use, deploy, and audit.

TREs should be built in such a way that they do not restrict or prevent interoperability where this may be desirable in future, by identifying and avoiding or removing barriers to interoperability.

Standardisation is also linked to the public trust principle, as a standard approach to TRE provision will make it easier for impacted parties to understand how their data will be used within TREs.

### Implications

> - Owing to the broad definition of {term}`TREs <trusted research environment (TRE)>`, there are no specific TRE technical standards.

Reviewer: ```suggestion
- Owing to the broad definition of {term}`TREs <trusted research environment (TRE)>`, there are currently no technical or information governance standards focused on TREs.
```

(satre_principles)=

# Architectural Principles

Architectural principles influence and shape the way you design and deliver a SATRE-aligned TRE.
They are a set of guiding considerations that sit above any specific architectural requirement, and can be applied across the entire architecture.

> Finally, the {ref}`TRE operator <infrastructure_roles>`s will need to consider different {ref}`roles <satre_roles>` with which individuals might interact with the TRE.
> 

Reviewer 2: ```suggestion
```

They consist of the following parts:

> **Statement**: A singular sentence that summarises the principle
> **Rationale**: Justification as to why this principle is important for the specification
> **Implications**: Things you need to consider or do to practise this principle

Reviewer 2: ```suggestion
:Statement: A singular sentence that summarises the principle
:Rationale: Justification as to why this principle is important for the specification
:Implications: Things you need to consider or do to practise this principle
```

(principle_usability)=

## Usability

### Statement

A TRE instance that works for all users minimises barriers to use, and provides a productive and accessible analysis environment for research.

### Rationale

There is often a trade-off between increased operational security and the usability of a TRE.
In order to maintain productivity, a TRE must balance these two competing aims.
The design and configuration of a TRE should allow all individuals involved with a TRE to effectively fulfil their roles.

### Implications

> - Robust TRE design and implementation should start by understanding users’ diverse expectations, needs, existing skillsets and preferences and responsibilities.

Reviewer 2: ```suggestion
- Robust TRE design and implementation should start by understanding users' diverse expectations, needs, existing skillsets and preferences and responsibilities.
```
We've got both unicode `’` and ascii `'` in this doc

  Any generated data should follow standards for provenance and transparency for audit trails.
- Different levels of observability may be needed for different users.
  Any data collected from an observability perspective should consider the needs of those who will use it, and minimise collection accordingly.
- There may be ethical and confidential issues to consider when implementing the observability principle.

(principle_standardisation)=

## Standardisation

### Statement

TREs should adhere to standards or well-known patterns wherever possible.

### Rationale

Standardisation makes it easier to design, operate, use and understand TREs, and reduces duplication of work.
This includes making TREs easier to use, deploy, and audit.

TREs should be built in such a way that they do not restrict or prevent interoperability where this may be desirable in future, by identifying and avoiding or removing barriers to interoperability.

Standardisation is also linked to the public trust principle, as a standard approach to TRE provision will make it easier for impacted parties to understand how their data will be used within TREs.

### Implications

> - Owing to the broad definition of {term}`TREs <trusted research environment (TRE)>`, there are no specific TRE technical standards.

Reviewer 2: Are we intentionally restricting this to technical (operators/developers) and not IG standards?

Reviewer: Updated
